### DISCUSSION TITLE
ListBox with ObservableCollection leaves first item visible after clearing the ObservableCollection

### DISCUSSION THREAD

**Discussion by sophia-anais** [AvaloniaUI/Avalonia#11226] [Q&A] [answered]
Hi,

I am binding an observable collection to my listbox. After clearing the listbox,  from time to time the first item of the listbox is still visible in my view, although my observable collection is not containing items anymore.

![image](https://user-images.githubusercontent.com/121035105/236152693-06bbeb0b-dcf5-4aa8-856b-20b58d3a4484.png)

I checked the visual and logical tree and found out, that all items are still listed there, but after clearing the listbox their property IsVisible is just changed to false. However for this one item, this property still remains true. Nevertheless it is not listed in the observable collection anymore.

![image](https://user-images.githubusercontent.com/121035105/236152349-a7517c82-0606-4c86-bd64-c4bccd5e78bd.png)

I am not interacting with the IsVisible-Property, just clearing my observable collection. Is there any way to prevent having this single item still being visible in my view?

**Comment by timunie** [comment, marked as answer]
1. Which Avalonia version? 
2. In the best case, can you file a simple sample to reproduce the issue? Because to me it sounds more like a bug we had during 11.0 updates.

**Reply by sophia-anais** [reply]
We used version 11.0.0-preview6. Having updated now to preview7, I can't reproduce the bug anymore, so it seems like the update solved the issue.

Thanks for pointing that out!
